### PR TITLE
changefeed(ticdc): add namespace info into ChangeFeedID

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: ref #5301

### What is changed and how it works?
make ChangefeedID to an struct
```
type ChangeFeedID struct {
	Namespace string
	ID        string
}
```

after this PR, all changefeed has an default namespace value, "default"
```
func DefaultNamespaceChangeFeedID(id string) ChangeFeedID {
	return ChangeFeedID{
		Namespace: DefaultNamespace,
		ID:        id,
	}
}
```

namesapce and ID  pair is unique in one ticdc cluster

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
Please add a release note.
 `None`.
```

### CODE REVIEW

File: cdc/model/changefeed.go [hunk 1]
```diff
@@ -31,8 +31,20 @@ import (
...
+const DefaultNamespace = "default"
+
 // ChangeFeedID is the type for change feed ID
-type ChangeFeedID = string
+type ChangeFeedID struct {
+	Namespace string
+	ID        string
+}
+
+func DefaultNamespaceChangeFeedID(id string) ChangeFeedID {
```
Comment (asddongmen) [inline cdc/model/changefeed.go:42]: `DefaultChangeFeedID`  is a more concise name.

File: cdc/model/changefeed.go [hunk 2]
```diff
@@ -31,8 +31,20 @@ import (
 	"go.uber.org/zap"
 )
 
+const DefaultNamespace = "default"
+
 // ChangeFeedID is the type for change feed ID
-type ChangeFeedID = string
+type ChangeFeedID struct {
+	Namespace string
```
Comment (asddongmen) [inline cdc/model/changefeed.go:38]: Can you add a comment for Namespce?

Comment (sdojjy) [reply cdc/model/changefeed.go:42]: done

File: cdc/contextutil/ctx.go [hunk 3]
```diff
@@ -109,31 +111,31 @@ func IsOwnerFromCtx(ctx context.Context) bool {
 
 // ChangefeedIDFromCtx returns a changefeedID stored in the specified context.
 // It returns an empty string if there's no valid changefeed ID found.
```
Comment (asddongmen) [inline cdc/contextutil/ctx.go:113]:
```suggestion
// It returns an empty model.changefeedID if there's no changefeedID found.
```

File: cdc/contextutil/ctx.go [hunk 4]
```diff
@@ -109,31 +111,31 @@ func IsOwnerFromCtx(ctx context.Context) bool {
...
+func ChangefeedIDFromCtx(ctx context.Context) model.ChangeFeedID {
+	changefeedID, ok := ctx.Value(ctxKeyChangefeedID).(model.ChangeFeedID)
 	if !ok {
-		return ""
+		return model.ChangeFeedID{}
 	}
 	return changefeedID
 }
 
 // PutChangefeedIDInCtx returns a new child context with the specified changefeed ID stored.
```
Comment (asddongmen) [inline cdc/contextutil/ctx.go:122]:
```suggestion
// PutChangefeedIDInCtx returns a new child context with the specified changefeedID stored.
```

**Review by asddongmen — APPROVED** [review]

**Pull request merged**